clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -6720,6 +6720,17 @@ void Clang::ConstructJob(Compilation &C, const JobAction &JA,
   Args.addOptOutFlag(CmdArgs, options::OPT_fassume_sane_operator_new,
                      options::OPT_fno_assume_sane_operator_new);
 
+  if (Args.hasFlag(options::OPT_fapinotes, options::OPT_fno_apinotes, false) ||
+      Args.hasArg(options::OPT_iapinotes_modules)) {
```

Comment:
If I am reading this correctly, this is not particularly helpful since it just repeats the conditions.

```
if (Args.hasFlag(options::OPT_fapinotes, options::OPT_fno_apinotes, false))
  CmdArgs.push_back("-fapinotes");
if (Args.hasArg(options::OPT_iapinotes_modules)) {
  if (Args.hasFlag(options::OPT_fapinotes_modules, options::OPT_fno_apinotes_modules, false))
    CmdArgs.push_back("-fapinotes-modules");
  Args.AddLastArg(CmdArgs, options::OPT_fapinotes_swift_version);
}
```

Or are the inner checks supposed to be for `OPT_fapinotes`?

Reply:
My mistake, the outer `if` condition became wrong during my refactoring attempt. Fixed it.

Reply:
Ah, now with the new version, it appears the double check is to deal with `-fapinotes-swift-version`?  I wonder if we can simplify this at all.  Is it invalid to pass along `-fapinotes-swift-version` if `-fapinotes` or `-fapinotes-modules` is not present?

Reply:
That's a good point, we don't actually need the outer check here. I removed it.